permission-portal-frontend/client/src/screens/CodeValidations.js

```diff
@@ -8,70 +7,62 @@ import { withStore } from '../store'
 import { Redirect } from 'react-router-dom'
 import { observer } from 'mobx-react'
 import PageTitle from '../components/PageTitle'
+import PendingOperationButton from '../components/PendingOperationButton'
 
 // snackbars docs can be found here:
 // https://material-ui.com/components/snackbars/
 
 const CodeValidationsBase = observer((props) => {
-  const [code, setCode] = useState('')
-  const [date, setDate] = useState('')
+  const [code, setCode] = useState('123-45-6')
+  const [toastInfo, setToastInfo] = useState({
+    success: false,
+    msg: '',
+  })
 
   let confirmedToast = createRef()
 
+  //TODO show confired toast when code confirmed by app
+
+  const genNewCode = async () => {
+    // TODO get new code from server
+    try {
```

Comment:
can use promises without try catch

Reply:
I'll add this when we know how we're going to talk to the server. Right now this method is just a placeholder